src/channel/ws/ws_0_8.ts

```diff
@@ -0,0 +1,656 @@
+/* eslint-disable no-underscore-dangle */
+import type {
+  BLOCK_HEADER,
+  EMITTED_EVENT,
+  NEW_TXN_STATUS,
+  SUBSCRIPTION_ID,
+  TXN_HASH,
+  TXN_WITH_HASH,
+} from '@starknet-io/starknet-types-08';
+
+import { BigNumberish, SubscriptionBlockIdentifier } from '../../types';
+import { JRPC } from '../../types/api';
+import { WebSocketEvent } from '../../types/api/jsonrpc';
+import { EventEmitter } from '../../utils/eventEmitter';
+import { TimeoutError, WebSocketNotConnectedError } from '../../utils/errors';
+import WebSocket from '../../utils/connect/ws';
+import { stringify } from '../../utils/json';
+import { bigNumberishArrayToHexadecimalStringArray, toHex } from '../../utils/num';
+import { Block } from '../../utils/provider';
+import { config } from '../../global/config';
+import { logger } from '../../global/logger';
+import { Subscription } from './subscription';
+
+/**
+ * Options for configuring the automatic reconnection behavior of the WebSocketChannel.
+ */
+export type ReconnectOptions = {
+  /**
+   * The number of retries to attempt before giving up.
+   * @default 5
+   */
+  retries?: number;
+  /**
+   * The initial delay in milliseconds before the first retry.
+   * This delay will be doubled for each subsequent retry (exponential backoff).
+   * @default 2000
+   */
+  delay?: number;
+};
+
+/**
+ * Options for configuring the WebSocketChannel.
+ */
+export type WebSocketOptions = {
+  /**
+   * The URL of the WebSocket endpoint of the Starknet node.
+   * @example 'ws://localhost:9545'
+   */
+  nodeUrl?: string;
+  /**
+   * This parameter can be used to provide a custom WebSocket implementation.
+   * This is useful in environments where the global WebSocket object is not available (e.g., Node.js).
+   * @example
+   * ```typescript
+   * import WebSocket from 'ws';
+   * const channel = new WebSocketChannel({ nodeUrl: '...', websocket: WebSocket });
```

Comment:
For this usage change where the `websocket` option is not an instance but a class/module the corresponding constructor initialization should be changed to something like this:
```
// type changes

nodeUrl: string;
websocket?: typeof WebSocket;
```
```
// constructor changes

this.nodeUrl = options.nodeUrl;
...
const ws = options.websocket || config.get('websocket') || WebSocket;
this.websocket = new ws(this.nodeUrl);
```